doc/source/client.rst

```diff
@@ -191,14 +191,9 @@ The logical devices represented by the device is addressed with the :mod:`slave=
 With **Serial**, the comm port is defined when creating the object.
 The physical devices are addressed with the :mod:`slave=` parameter.
 
-:mod:`slave=0` is used as broadcast in order to address all devices.
-However experience shows that modern devices do not allow broadcast, mostly because it is
-inheriently dangerous. With :mod:`slave=0` the application can get upto 254 responses on a single request,
-and this is not handled with the normal API calls!
-
-The simple request calls (mixin) do NOT support broadcast, if an application wants to use broadcast
-it must call :mod:`client.execute` and deal with the responses.
+:mod:`slave=0` is used as broadcast in order to address all devices. To accommodate non-standard behaviour of devices, all request calls to :mod:`slave=0` will expect and wait for at least one response. If multiple devices on the bus are expected to respond to broadcast requests (for example for the purpose of device detection), the application can get upto 254 responses on a single request, and this is not handled with the normal API calls. If an application is expecting multiple responses to a broadcast request, it must call :mod:`client.execute` and deal with the responses.
 
+If no response is expected to a broadcast or unicast write request, the :mod:`no_response_expected=True` argument can be used in the normal write API calls. However, these write operations will not return indication of failure or success, and will not respect the turnaround delay after sending a broadcast message, therefore, it is the application's responsibility to allow time for the slaves to process the broadcast request before the next request is made on the bus.
```

Comment:
to a slave=0 request (remove unicast, that I think is not defined in Modbus).

We should add the parameter to all calls and not just write calls, that way the API is uniform, when the pdu is refactored, we might support read with slave=0 in the standard API.

Reply:
Will update. Just to make sure I'm not getting informed from the wrong source, I have been referring to the specification laid out in this document: https://www.modbus.org/docs/Modbus_over_serial_line_V1_02.pdf
which is labelled "Please use this for new MODBUS implementations."

If this is no longer applicable or valid, please let me know